### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by CunliangGeng**
Major changes
- remove unused dependencies and add required ones
- fix typing error in python v3.9 by adding `from __future__ import annotations`
- fix bug on checking gnps website status

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#229** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/229?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#228** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/228?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#227** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/227?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#226** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/226?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#224** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/224?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: requirements.dev.txt [hunk 1]
```diff
@@ -1,22 +1,23 @@
-autoflake
+# packaging
```
Comment (gcroci2) [inline requirements.dev.txt:1]: Why we don't use extra dependencies list `dev` in the TOML file? Why a separate txt file is needed? Same question for requirements.txt

File: src/nplinker/metabolomics/gnps/gnps_downloader.py [hunk 2]
```diff
@@ -53,7 +58,7 @@ def gnps_format(self) -> GNPSFormat:
         """
         return self._gnps_format
 
-    def download(self) -> Self:
+    def download(self) -> "Self":
```
Comment (gcroci2) [inline src/nplinker/metabolomics/gnps/gnps_downloader.py:61]: Why are the double quotation marks needed?

**Review by gcroci2 — APPROVED** [review]
I have only a couple of questions but I already approve it :)

File: src/nplinker/metabolomics/gnps/gnps_downloader.py (hunk 2 above)
Comment (CunliangGeng) [reply src/nplinker/metabolomics/gnps/gnps_downloader.py:61]: Without the double quote, there is always a warning on `Self` in vscode, it looks vscode cannot recognise it.

File: requirements.dev.txt (hunk 1 above)
Comment (CunliangGeng) [reply requirements.dev.txt:1]: That's for historical reasons. It looks putting everything in `pyproject.toml` is easier to manage. I'll change it in a new commit.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Mar 21, 8:44 AM EDT**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/226).
* **Mar 21, 8:45 AM EDT**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/226).

**Pull request merged**